controllers/crd_controller.go

```diff
@@ -85,5 +106,20 @@ func (r *CRDReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	if err != nil {
 		return err
 	}
-	return c.Watch(&source.Kind{Type: &apiv1.CustomResourceDefinition{}}, &handler.EnqueueRequestForObject{})
+	config, err := buildKubeConnectionConfig()
```

Comment:
Why the need to extract this function?

Reply:
Just for the better function name... but yeah you are right we don't need to extract... used the internal function directly